flocker/node/agents/test/test_blockdevice.py

```diff
@@ -698,6 +752,186 @@ def add_application_with_volume(node_state):
                                   mountpoint=FilePath(b"/data")))})
 
 
+_BLOCKDEVICE_PREFIX = "blockdevice-"
+
+
+def _get_dataset_id_for_test_blockdevice_id(test_blockdevice_id):
+    """
+    Determine the dataset_id that was used to create a test_blockdevice_id. For
+    use with _create_test_blockdevice_id_for_dataset_id for validation of
+    opaque test blockdevice_ids with the dataset_ids they were constructed
+    from.
+
+    :param unicode test_blockdevice_id: test blockdevice_id returned by
+        _create_test_blockdevice_id_for_dataset_id.
+
+    :returns: The dataset_id used to create the test blockdevice_id.
+    """
+    return test_blockdevice_id[len(_BLOCKDEVICE_PREFIX):]
+
+
+def _create_test_blockdevice_id_for_dataset_id(dataset_id):
```

Comment:
Doesn't need to be separate function. Also can probably just do `_BLOCKDEVICE_PREFIX + dataset_id`.